server/src/test/definition/ModelDefinitionProvider.spec.ts

```diff
@@ -0,0 +1,123 @@
+import * as fs from 'fs';
+import { assertThat } from 'hamjest';
+import { Position, Range } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { DbtRepository } from '../../DbtRepository';
+import { DbtDefinitionProvider } from '../../definition/DbtDefinitionProvider';
+import { ModelDefinitionProvider } from '../../definition/ModelDefinitionProvider';
+
+describe('ModelDefinitionProvider', () => {
+  const PATH_TO_PROJECT = '/Users/user_name/project';
+  const FILE_NAME = `model_with_refs.sql`;
+
+  const PACKAGE_NAME = 'package';
+  const PACKAGE_MODEL = 'package_model';
+  const PROJECT_NAME = 'project';
+  const PROJECT_MODEL = 'project_model';
+
+  const PACKAGE_MODEL_ROOT_PATH = `dbt_packages/${PACKAGE_NAME}/models/${PACKAGE_NAME}.sql`;
+  const PROJECT_MODEL_ROOT_PATH = `models/${PROJECT_MODEL}.sql`;
+
+  const DBT_REPOSITORY = new DbtRepository();
+  const PROVIDER = new ModelDefinitionProvider(DBT_REPOSITORY);
+
+  let modelDocument: TextDocument;
+
+  before(() => {
+    DBT_REPOSITORY.models = [
+      {
+        uniqueId: `model.${PACKAGE_NAME}.${PACKAGE_MODEL}`,
+        rootPath: PATH_TO_PROJECT,
+        originalFilePath: PACKAGE_MODEL_ROOT_PATH,
+        name: PACKAGE_MODEL,
+        packageName: PACKAGE_NAME,
+        database: '',
+        schema: '',
+        dependsOn: { nodes: [] },
+        refs: [],
+      },
+      {
+        uniqueId: `model.${PROJECT_NAME}.${PROJECT_MODEL}`,
+        rootPath: PATH_TO_PROJECT,
+        originalFilePath: PROJECT_MODEL_ROOT_PATH,
+        name: PROJECT_MODEL,
+        packageName: PROJECT_NAME,
+        database: '',
+        schema: '',
+        dependsOn: { nodes: [] },
+        refs: [],
+      },
+    ];
+
+    const modelFileContent = fs.readFileSync(`${__dirname}/../../../src/test/sql_files/${FILE_NAME}`, 'utf8');
+    modelDocument = TextDocument.create('/path/to/model', 'sql', 0, modelFileContent);
+  });
+
+  it('provideDefinitions should return definition for model from package (if package is specified)', () => {
+    assertThat(
+      PROVIDER.provideDefinitions(modelDocument, Position.create(1, 31), {
+        value: modelDocument.getText(Range.create(1, 5, 1, 42)),
```

Comment:
May be we can just write here `"{{ ref('package', 'package_model') }}"`? Probably this will help to understand this test more easily.